content/developer/howtos/standalone_owl_application.rst

```diff
@@ -0,0 +1,176 @@
+
+To keep things simple, let's start with a very straightforward component that just renders
+"Hello, World!". This will let us know at a glance if our setup is working.
+
+First, create the template in `/your_module/static/src/standalone_app/root.xml`.
+
+.. code-block:: xml
+
+    <?xml version="1.0" encoding="UTF-8"?>
+    <templates xml:space="preserve">
+        <t t-name="your_module.Root">
+            Hello, World!
+        </t>
+    </templates>
+
+Then create the JavaScript file for that component in `/your_module/static/src/standalone_app/root.js`.
+
+.. code-block:: js
+
+    /** @odoo-module */
+    import { Component } from "@odoo/owl";
+
+    export class Root extends Component {
+        static template = "your_module.Root";
+    }
+
+.. seealso::
+   Owl components :ref:`reference<frontend/components>`.
+
+2. Setting up the environment and mounting the application
+==========================================================
+
+An Owl application needs an environment, and the Odoo JavaScript framework needs that environment to
+contain the :ref:`services<frontend/services>`.
+Starting the services is also required to load the translations, which we need to do before mounting
+our Owl application, so that we can give Owl a working translation function.
+
+Create the JavaScript file that will mount the app in `/your_module/static/src/standalone_app/app.js`.
+
+.. code-block:: js
+
+    /** @odoo-module */
+    import { whenReady, App } from "@odoo/owl";
+    import { makeEnv, startServices } from "@web/env";
+    import { templates } from "@web/core/templates";
+    import { _t } from "@web/core/l10n/translation";
+    import { Root } from "@your_module/standalone_app/root";
+
+    whenReady().then(async () => {
+        const env = makeEnv();
+        await startServices(env);
+        const app = new App(Root, {
+            env,
+            templates,
+            dev: env.debug,
+            translateFn: _t,
+            translatableAttributes: ["data-tooltip"],
+        });
+        return app.mount(document.body);
+    });
+
+3. Creating an assets bundle containing our code
+================================================
+
+In the manifest of your module, create a new :ref:`assets bundle<reference/assets_bundle>`.
+It should include the `web._assets_core` bundle, which contains the Odoo JavaScript
+framework and the core libraries it needs (e.g. Owl and luxon), after which you can have a
+glob that adds all the files for your application in the bundle.
+
+.. code-block:: py
+
+    {
+        # ...
+        'assets': {
+            'your_module.assets_standalone_app': [
+                ('include', 'web._assets_helpers'),
+                'web/static/src/scss/pre_variables.scss',
+                'web/static/lib/bootstrap/scss/_variables.scss',
+                ('include', 'web._assets_bootstrap'),
+                ('include', 'web._assets_core'),
+                'your_module/static/src/standalone_app/**/*',
```

Comment:
just one last comment: I get why you put the content of the app in a sub folder, but maybe we should simply do it in `src`. In most case, it will be fine. Not sure though, if you prefer `src/standalone_app`, i can live with it

Reply:
I think this avoids the pitfall of people including their entire src folder on their controller when their module already has some files in static that they use to customize the webclient. Maybe I should add a warning in that spirit so that people don't include this folder in the assets backend either though.